discojs-web/src/memory/memory.ts

```diff
@@ -10,62 +10,72 @@ import { Map } from 'immutable'
 import * as tf from '@tensorflow/tfjs'
 
 import type { Path, Model, ModelInfo, ModelSource } from '@epfml/discojs'
-import { Memory, ModelType, models } from '@epfml/discojs'
+import { Memory, StoredModelType, models } from '@epfml/discojs'
 
 export class IndexedDB extends Memory {
-  override pathFor (source: ModelSource): Path {
+  override getModelMemoryPath (source: ModelSource): Path {
     if (typeof source === 'string') {
       return source
     }
-
-    if (source.type === undefined || source.taskID === undefined || source.name === undefined) {
+    if (source.type === undefined || source.taskID === undefined ||
+      source.name === undefined || source.tensorBackend === undefined) {
       throw new TypeError('source incomplete')
     }
-
     const version = source.version ?? 0
-
-    return `indexeddb://${source.type}/${source.taskID}/${source.name}@${version}`
+    return `indexeddb://${source.type}/${source.tensorBackend}/${source.taskID}/${source.name}@${version}`
   }
 
-  override infoFor (source: ModelSource): ModelInfo {
+  override getModelInfo (source: ModelSource): ModelInfo {
     if (typeof source !== 'string') {
       return source
     }
-    const [stringType, taskID, fullName] = source.split('/').splice(2)
+    const [stringType, tensorBackend, taskID, fullName] = source.split('/').splice(2)
 
-    const type = stringType === 'working' ? ModelType.WORKING : ModelType.SAVED
+    const type = stringType === 'working' ? StoredModelType.WORKING : StoredModelType.SAVED
 
     const [name, versionSuffix] = fullName.split('@')
     const version = versionSuffix === undefined ? 0 : Number(versionSuffix)
-    return { type, taskID, name, version }
+    if (tensorBackend != 'tfjs' && tensorBackend != 'gpt') {
```

Comment:
js is quite bad at comparing stuff, it loves to convert things if types differ. that's why `===` & `!==` where introduce, to not do that, as it creates unexcepted bugs in some cases. for example `{ toString: () => 'asd' } == 'asd'` is true which might come at a surprise.
anyway, I think that explicit is way more readable than implicit, or if the latter is used, a comment is welcome.

```suggestion
    if (tensorBackend !== 'tfjs' && tensorBackend !== 'gpt') {
```

Reply:
Ah yes indeed